### DISCUSSION TITLE
Google Search Console

### DISCUSSION THREAD

**Discussion by BlaineMoore** [plausible/analytics#3046] [Self-Hosted Support] [answered]
Is there a way to integrate Google Search Console on the self-hosted version? I just get an error icon instead of the "Continue with Google" button...

![image](https://github.com/plausible/analytics/assets/8483773/bd71bc2e-bcab-438b-b752-abcee2343787)

**Comment by ruslandoga** [comment]
👋 @BlaineMoore 

Have you tried following the instructions in the link? If so and it didn't work, maybe the integration guide needs to be updated.

**Reply by ruslandoga** [reply, marked as answer]
These two documents seem relevant:

- https://plausible.io/docs/self-hosting-configuration#google-search-integration
- https://plausible.io/docs/google-search-console-integration

**Reply by ruslandoga** [reply]
This message seems to be shown when `GOOGLE_CLIENT_ID` is not set.

- https://github.com/plausible/analytics/blob/c19a51efb0dbe2da0f8dede8e3d5df167bd7fba1/lib/plausible_web/templates/site/settings_search_console.html.eex#L10
- https://github.com/plausible/analytics/blob/c19a51efb0dbe2da0f8dede8e3d5df167bd7fba1/lib/plausible_web/templates/site/settings_general.html.eex#L74

**Reply by BlaineMoore** [reply]
Ahh, I saw this one: https://plausible.io/docs/google-search-console-integration

I'd missed the google console directions under the configuration options page, though. Thanks, that was the issue, I hadn't set it in the environment ahead of time. The Google Console specific page doesn't mention that (which in retrospect makes sense.)

Thanks!
